packages/svelte/src/reactivity/map.js

```diff
@@ -102,12 +103,10 @@ export class SvelteMap extends Map {
 			increment(version);
 		} else if (prev_res !== value) {
 			increment(s);
-			// If no one listening to this property and is listening to the version, or
-			// the inverse, then we should increment the version to be safe
-			if (
-				(s.reactions === null && version.reactions !== null) ||
-				(s.reactions !== null && version.reactions === null)
-			) {
+
+			// if not every reaction of s is a reaction of version we need to also include version
+			const needs_version_increase = !s.reactions?.every((r) => version.reactions?.includes(r));
```

Comment:
If you make a `Set` out of `version.reactions` then this will only be an O(n) operation.

Reply:
I thought of this but didn't want to touch too much without validating this idea first...do you think it's the right approach?

Reply:
Yes, I actually thought about doing this when I first implemented the fix. Then I couldn't find I test case where it mattered, so I went for the simpler approach. So clearly the more complex was needed after-all. I used a Set to improve the performance – but only if `version.reactions` was not null, otherwise I did a short path that avoided any iterations entirely.

Reply:
Ok let me do it now

Reply:
Sorry, I didn't mean changing `reactions` to a Set, I meant locally making a Set of the array and using that.

Reply:
I was thinking:

```js
var v_reactions = version.reactions === null ? null : new Set(version.reactions)
var needs_version_increase = v_reactions === null || !s.reactions?.every((r) => v_reactions.has(r));
```

Reply:
Ohhh gotcha